### PR TITLE
Adds ability to restore from a previous backup

### PR DESCRIPTION
**Pull request by jathak**
I've implemented `RestoreProtocol` based on Option B (as discussed on #81) as it seemed to strike the right balance between preserving student data and convenience.

### CODE REVIEW

File: client/protocols/restore.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if not self.args.restore:
+            return
+        
+        self.access_token = auth.authenticate(self.args.authenticate)
+        
+        print('Loading backups...')
+        
+        response = self.request('user')
+        if not response:
+            print('Could not connect to server')
```
Comment (albert12132) [inline client/protocols/restore.py]: Should this `return` to exit out of the function?

File: client/protocols/restore.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        user = response['data']['results'][0]
+        email = user['email'][0]
+        assign_id = self.get_assign_id(self.assignment.endpoint)
+        params = {
```
Comment (albert12132) [inline client/protocols/restore.py]: Where is this variable used?

File: client/protocols/restore.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        current_time = datetime.now()
+        print('0: Cancel Restore')
+        for current in range(0, len(backups)):
+            backup = backups[current]
+            time_diff = get_time_diff(backup['timestamp'], current_time)
+            print('{0}: {1} by {2}'.format(current + 1, time_diff, backup['submitter']))
+        response = input('Backup #: ')
+        selection = int(response) - 1
+        if selection < 0 or selection >= len(backups):
+            return
```
Comment (albert12132) [inline client/protocols/restore.py]: Should probably print out a message saying "Invalid option for restoring backups."

**Comment by albert12132** [comment]
Made some comments. Otherwise, LGTM.

**Comment by jathak** [comment]
Fixed those. I also added this to the default config.ok file, since I forgot to previously.

**Comment by albert12132** [comment]
Made one more small comment about the config file. Looks good to merge once it's been fixed.

**Comment by jathak** [comment]
Whoops. Fixed.

**Comment by albert12132** [comment]
LGTM.

**Pull request merged**